### PR TITLE
Support Windows npipe.

### PR DESCRIPTION
**Pull request by calavera**
So docker plugins and other libraries can connect with Windows daemons.

Signed-off-by: David Calavera <david.calavera@gmail.com>

### CODE REVIEW

File: sockets/sockets.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+// ConfigureTransport configures the specified Transport according to the
+// specified proto and addr.
+// If the proto is unix (using a unix socket to communicate) the compression
+// is disabled.
+func ConfigureTransport(tr *http.Transport, proto, addr string) {
+	// Why 32? See https://github.com/docker/docker/pull/8035.
+	timeout := 32 * time.Second
+	if proto == "unix" {
```
Comment (LK4D4) [inline sockets/sockets.go]: time for switch-case

File: sockets/sockets.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"time"
+)
+
+// ConfigureTransport configures the specified Transport according to the
+// specified proto and addr.
+// If the proto is unix (using a unix socket to communicate) the compression
+// is disabled.
+func ConfigureTransport(tr *http.Transport, proto, addr string) {
+	// Why 32? See https://github.com/docker/docker/pull/8035.
+	timeout := 32 * time.Second
```
Comment (LK4D4) [inline sockets/sockets.go]: should be const I think.

**Comment by calavera** [comment]
@LK4D4 `time.Duration` cannot be constant afaik. I made a change to `switch`, thanks for the suggestion.

File: sockets/sockets.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+func ConfigureTransport(tr *http.Transport, proto, addr string) {
+	switch proto {
+	case "unix":
+		// No need for compression in local communications.
+		tr.DisableCompression = true
+		tr.Dial = func(_, _ string) (net.Conn, error) {
+			return net.DialTimeout(proto, addr, defaulTimeout)
+		}
+	case "npipe":
+		tr.Dial = func(_, _ string) (net.Conn, error) {
```
Comment (LK4D4) [inline sockets/sockets.go]: what about DisableCompression here as well? ping @jhowardmsft

**Comment by LK4D4** [comment]
@calavera it definitely can be const. It's just int64.

**Comment by calavera** [comment]
@LK4D4 fixed.

**Comment by LK4D4** [comment]
LGTM
However, I still want to hear about compression from windows guys. It seems unnecessary for npipe.

**Comment by jhowardmsft** [comment]
@jstarks re compression

**Comment by jstarks** [comment]
Good point, might as well turn off compression. Probably we should follow up on engine-api and do the same there. I somehow forgot to do that when I made the change there.

**Comment by calavera** [comment]
Thanks @jstarks!

I'm going to update engine-api to use this function directly, so we don't need to have this duplicated.

**Comment by calavera** [comment]
@LK4D4 updated to disable compression.

**Comment by LK4D4** [comment]
@calavera However you're the only maintainer here :) But it's lgtm.

**Comment by calavera** [comment]
technical details :wink:

**Pull request merged**